Fix: Make Collections Public on Creation

Chickensoupwithrice opened the pull request:
Resolves #1212 

- fix: #1212 Add isPublic to Add Collection endpoint
- fix: frontend sends `public` not `isPublic`

The model was previously wrong, so I updated it, but I get the sense that the frontend should actually use `isPublic` instead of `public`. For example, the PATCH endpoint currently does ensure that a collection will be public if set so through the use of the `isPublic` variable on the request.

Tested locally and creation leads to a new publically accessible collection.

Chickensoupwithrice committed:
fix: #1212 Add isPublic to Add Collection endpoint

Chickensoupwithrice committed:
fix: frontend sends `public` not `isPublic`

Chickensoupwithrice committed:
fix: linting

Chickensoupwithrice commented:
This is the frontend request sending `public` when creating a new collection:

![image](https://github.com/webrecorder/browsertrix-cloud/assets/22575913/58d03f20-0240-4c82-889a-ecd88fe59ffe)

And this is the PATCH edition, which correctly sends `isPublic`
![image](https://github.com/webrecorder/browsertrix-cloud/assets/22575913/a6c33a5c-bffe-41ab-93ec-a6fb972d1299)

ikreymer commented on backend/btrixcloud/models.py:
```diff
@@ -506,7 +506,7 @@ class CollIn(BaseModel):
     description: Optional[str]
     crawlIds: Optional[List[str]] = []
 
-    isPublic: Optional[bool] = False
```
~~What's the reason for this change? Perhaps should just keep as is to be consistent?~~

Edit: just saw rest of the comment, I think we should keep this and instead change frontend here to also use isPublic

Chickensoupwithrice replied:
I'm following what the frontend sends, so I had to change the model. I attached screenshots in the comments about how this is inconsistent, but this currently accurately represents the data of the request.

ikreymer commented:
Can you also add/modify a test in backend/test/test_collections.py

Chickensoupwithrice committed:
fix: make change in the frontend

Chickensoupwithrice committed:
feat: add test for isPublic on creation

Chickensoupwithrice commented:
Changed the frontend instead to be consistent with the rest of the application. 
Also added a test to ensure that public created collections are in fact public.

tw4l commented on backend/test/test_collections.py:
```diff
@@ -48,6 +49,31 @@ def test_create_collection(
+        f"{API_PREFIX}/orgs/{default_org_id}/collections/{_public_coll_id}",
+        headers=crawler_auth_headers,
+    )
+    assert r.json()["isPubilc"] == True
```
```suggestion
    assert r.json()["isPublic"]
```

assert will ensure it's True :)

tw4l commented on backend/btrixcloud/colls.py:
```diff
@@ -386,7 +388,11 @@ async def add_collection(
+            new_coll.name,
+            new_coll.crawlIds,
+            new_coll.description,
+            new_coll.isPublic,
```
Might be better to pass in the whole new_coll `CollIn` object here rather than including all the separate fields

tw4l commented on backend/btrixcloud/colls.py:
```diff
@@ -65,6 +65,7 @@ async def add_collection(
         name: str,
         crawl_ids: Optional[List[str]],
         description: Optional[str] = None,
+        is_public: Optional[bool] = False,
```
See other comment, we can reduce this to just `new_coll: CollIn` and then set the values in the `Collection` init accordingly

ikreymer commented on frontend/src/pages/org/collections-new.ts:
```diff
@@ -68,7 +68,7 @@ export class CollectionsNew extends LiteElement {
             description,
             crawlIds,
-            public: isPublic,
+            isPublic: isPublic,
```
can just be:
```suggestion
            isPublic,
```

Chickensoupwithrice committed:
Update frontend/src/pages/org/collections-new.ts

Co-authored-by: Ilya Kreymer <ikreymer@users.noreply.github.com>

Chickensoupwithrice committed:
Update backend/test/test_collections.py

Co-authored-by: Tessa Walsh <tessa@bitarchivist.net>

Chickensoupwithrice committed:
fix(colls): pass in CollIn in it's entirety

Chickensoupwithrice committed:
fix(backend tests): Assert values of 3

ikreymer commented on backend/test/test_collections.py:
```diff
@@ -1,3 +1,4 @@
+from pickle import PickleBuffer
```
typo / extra line added?

Chickensoupwithrice replied:
Oops yeah, I think aggressive LSP imported it by mistake

ikreymer commented on backend/test/test_collections.py:
```diff
@@ -1,3 +1,4 @@
+from pickle import PickleBuffer
```
```suggestion
from pickle import PickleBuffer
```

ikreymer committed:
remove unused line

ikreymer committed:
reformat
